### DISCUSSION TITLE
Id field missing from CreateInput

### DISCUSSION THREAD

**Discussion by ZosoDerGoldene** [prisma/prisma#9443] [Q&A] [answered]
I have a create statement with a couple of nested create queries nested inside (and create queries nested in those).
Although the model would theoretically allow prisma to automatically fill the respective id fields, that fails starting level two, so I tried switching to pregenerated uuid and explicitely using those in the nested queries - because the id field is missing from xyzCreateInput, it's only part of xyzUncheckedCreateInput. That is – counterintuitive –, so I can't help but wonder what the reason for such a limitation would be.

For me, one of the main use cases for explicitely providing an id (apart from client-generated ids and migration) doesn't work. Any insights? It sure feels like a bug, but I definitely don't know enough make that call.

**Comment by ZosoDerGoldene** [comment, marked as answer]
Figured it out, it would create circular references

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
